### PR TITLE
Added add/remove broker endpoints to the Cruise Control API

### LINKED ISSUE #6737: Adding `add_broker` and `remove_broker` endpoints handling to Cruise Control API
Following the proposal ["Extending the KafkaRebalance resource with rebalance types to help scaling the Apache Kafka cluster"](https://github.com/strimzi/proposals/pull/52), there is the need for extending the Cruise Control API in the Strimzi cluster operator by handling the two following endpoints:

* [add_broker](https://github.com/linkedin/cruise-control/wiki/REST-APIs#add-a-list-of-new-brokers-to-kafka-cluster)
* [remove_broker](https://github.com/linkedin/cruise-control/wiki/REST-APIs#decommission-a-list-of-brokers-from-the-kafka-cluster)

### PR DESCRIPTION
**Pull request by ppatierno**
### Type of change

- Enhancement / new feature

### Description

As first step of proposal [#35](https://github.com/strimzi/proposals/blob/main/035-rebalance-types-scaling-brokers.md), this PR fixes #6737  and adds the handling of `add_broker` and `remove_broker` endpoints to the Cruise Control API implementation in the cluster operator.
These new endpoints behave the same way as the `/rebalance` one which is used today for a full-rebalance across all the brokers in the cluster. 
All of rebalance related endpoints have most of the parameters in common. Taking a look at the Cruise Control wiki there are a few specific parameters for each endpoint but they are not handled right now. Anyway the rebalance options are defined as a hierarchy.
Other than adding the handling of the endpoints, this PR does some refactoring around the implementation and the corresponding tests. This is always because the rebalance related endpoints have common behaviour, so I avoided to have code duplicated.

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [x] Write tests
- [x] Make sure all tests pass
- [x] Reference relevant issue(s) and close them after merging

### CODE REVIEW

File: cluster-operator/pom.xml
Comment (scholzj) [inline cluster-operator/pom.xml:20]: This is not new ... but why is this in the Cluster Operator pom.xml?

**Review by scholzj — APPROVED** [review]
LGTM. It should be reviewed by SMEs.

File: cluster-operator/pom.xml
Comment (tomncooper) [reply cluster-operator/pom.xml:20]: I suppose this property could be moved to the parent? It is only used by the cluster operator tests though?

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/cruisecontrol/CruiseControlApi.java
Comment (tomncooper) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/cruisecontrol/CruiseControlApi.java:54]: Should this not be `addBrokers` (plural) as you can add more than one.

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/cruisecontrol/CruiseControlApi.java
Comment (tomncooper) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/cruisecontrol/CruiseControlApi.java:68]: As above, this method name should be plural.

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/cruisecontrol/CruiseControlApiImpl.java
Comment (tomncooper) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/cruisecontrol/CruiseControlApiImpl.java:88]: I don't really have a problem with changing the name, but I just wonder why?

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/cruisecontrol/CruiseControlApiImpl.java
Comment (tomncooper) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/cruisecontrol/CruiseControlApiImpl.java:244]: Plural method name?

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/cruisecontrol/CruiseControlApiImpl.java
Comment (tomncooper) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/cruisecontrol/CruiseControlApiImpl.java:263]: Plural method name?

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/cruisecontrol/PathBuilder.java
Comment (tomncooper) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/cruisecontrol/PathBuilder.java:98]: You know what I am gonna say.........plural "Brokers"?

File: cluster-operator/src/test/resources/io/strimzi/operator/cluster/operator/assembly/CruiseControlJSON/CC-Broker-not-exist.json
Comment (tomncooper) [inline cluster-operator/src/test/resources/io/strimzi/operator/cluster/operator/assembly/CruiseControlJSON/CC-Broker-not-exist.json:2]: Doesn't this also apply to the /remove_broker endpoint as well?

**Review by tomncooper — APPROVED** [review]
LGTM. Only nit was the method names, I think they should be plural as you can add/remove more than one broker. But the CC endpoint uses the singular so I will leave it up to you.

**Review by kyguy — APPROVED** [review]
Nice work Paolo!

File: cluster-operator/pom.xml
Comment (ppatierno) [reply cluster-operator/pom.xml:20]: Yeah it's used only here but maybe it's better to move to the parent to have all version definitions in one place.

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/cruisecontrol/CruiseControlApi.java
Comment (ppatierno) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/cruisecontrol/CruiseControlApi.java:54]: I had this doubt for days :-D ... but I decided to reflect the name of the `/add_broker` endpoint and keeping the API the same.

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/cruisecontrol/CruiseControlApi.java
Comment (ppatierno) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/cruisecontrol/CruiseControlApi.java:68]: Ditto as above.

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/cruisecontrol/CruiseControlApiImpl.java
Comment (ppatierno) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/cruisecontrol/CruiseControlApiImpl.java:88]: Personally but even in general I see a "builder" made by `withXXXX` methods (as we are using in the rebalance options for example). Was there any specific reasons you used `addXXXX`?

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/cruisecontrol/CruiseControlApiImpl.java
Comment (ppatierno) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/cruisecontrol/CruiseControlApiImpl.java:244]: Ditto as above.

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/cruisecontrol/CruiseControlApiImpl.java
Comment (ppatierno) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/cruisecontrol/CruiseControlApiImpl.java:263]: Ditto as above.

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/cruisecontrol/CruiseControlRebalanceResponse.java
Comment (ShubhamRwt) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/cruisecontrol/CruiseControlRebalanceResponse.java:22]: @ppatierno Hi, not any change by you but seems like there is a typo here `isProposalStillCalaculating.

**Review by ShubhamRwt — APPROVED** [review]

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/cruisecontrol/CruiseControlRebalanceResponse.java
Comment (ppatierno) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/cruisecontrol/CruiseControlRebalanceResponse.java:22]: Yeah thanks. I will leave the regression passing and then making the change before merging (it won't have impact on regression again).

Comment (ppatierno) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/cruisecontrol/CruiseControlRebalanceResponse.java:22]: Damn merged the PR before fixing this, going to open a trivial new one.

**Pull request merged**
